src/gluonts/core/serde.py

```diff
@@ -206,11 +206,18 @@ def _dump_code(x: Any) -> str:
         elif isinstance(x, tuple):
             elems = [dump_code(v) for v in x]
             return "(" + ", ".join(elems) + ",)"
+        elif isinstance(x, set):
+            elems = [dump_code(v) for v in x]
+            return "{x" + ", ".join(elems) + ",}"
```

Comment:
Does this ever trigger? Since `x` is coming from encode it should never be a set.

Reply:
You are correct, it doesn't trigger. Forgot that `encode` is called before serializing in `code` format.